ops/charm.py

```diff
@@ -364,15 +362,19 @@ class RelationEvent(HookEvent):
     "stopped". Within that time window, the unit may participate in
     several different relations at a time, including multiple
     relations with the same name.
+    """
 
-    Attributes:
-        relation: The :class:`~ops.model.Relation` involved in this event
-        app: The remote :class:`~ops.model.Application` that has triggered this
-             event
-        unit: The remote :class:`~ops.model.Unit` that has triggered this event. This may be
-              ``None`` if the relation event was triggered as an
-              :class:`~ops.model.Application` level event
+    relation: 'Relation'
+    """The relation involved in this event."""
+
+    app: Optional[model.Application]
+    """The remote application that has triggered this event."""
```

Comment:
maybe mention what it means if the app is None? with this wording the Optional feels very strange. Same for `unit` below

Reply:
Hmm, I think `app` should probably never be None. However, I struggled to confirm that quickly, so I've done the lazy thing and added a TODO comment to look into this later.